apps/frontend/src/pages/quiz/libs/components/quiz-form/quiz-form.tsx

```diff
@@ -59,14 +62,78 @@ const QuizForm: React.FC<Properties> = ({ onNext }: Properties) => {
 		void dispatch(quizActions.previousQuestion());
 	}, [dispatch]);
 
-	const handleNextStep = useCallback(() => {
-		if (isLast) {
+	useEffect(() => {
+		if (categoryDone.includes(currentCategoryIndex)) {
+			setIsDisabled(false);
+		}
+	}, [categoryDone, currentCategoryIndex]);
+
+	const handleOnChange = useCallback(() => {
+		if (!category) {
 			return;
 		}
 
-		void dispatch(quizActions.nextQuestion());
-		reset();
-	}, [dispatch, isLast, reset]);
+		const questionLabels = category.map(
+			(categoryItem) => `question${categoryItem.id.toString()}`,
+		);
+
+		const formValues = getValues();
+
+		const hasUndefined = questionLabels.some(
+			(question) => formValues[question] === undefined,
+		);
+
+		if (!hasUndefined) {
+			setCategoryDone([...categoryDone, currentCategoryIndex]);
+			setIsDisabled(false);
+		}
+	}, [category, categoryDone, currentCategoryIndex, getValues]);
+
+	const getAnswerIds = useCallback((formData: FormToSave) => {
+		return Object.values(formData).map(Number);
+	}, []);
+
+	const handleNextStep = useCallback(
+		(data: QuizFormValues) => {
+			if (!category) {
+				return;
+			}
+
+			const questionLabels = category.map(
+				(categoryItem) => `question${categoryItem.id.toString()}`,
+			);
+			const hasUndefinedAnswers = questionLabels.some(
+				(question) => data[question] === undefined,
+			);
+
+			if (!hasUndefinedAnswers) {
+				const questionFormAnswers = Object.fromEntries(
+					Object.entries(data).filter(([key]) => key !== "answers"),
+				);
+
+				if (isLast) {
+					const answerIds = getAnswerIds(questionFormAnswers);
+
+					void dispatch(quizActions.saveAnswers({ answerIds }));
+
+					onNext();
+				}
+
+				setCategoryDone([...categoryDone, currentCategoryIndex]);
```

Comment:
Why category? Isn't it quiz question?

Reply:
There are 3 questions in 1 category so I was thinking it'll be more ideal to use category